app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -217,11 +217,11 @@ public function prefetchEstimateAction()
 
             // ----------^_^----------- //
             $shippingAddress = array(
-                'city'       => @$shippingAddressOriginal['city'],
-                'region'     => @$shippingAddressOriginal['region'],
+                'city'       => @$shippingAddressOriginal['city'] ? $shippingAddressOriginal['city'] : null,
```

Comment:
I'm PHP noob. Isn't `@$shippingAddressOriginal['city']` return `null` if 'city' does not exist?

Reply:
@daisy1754 yes, you are right if would return null if `$shippingAddressOriginal['city']` does not exsit when try to throw exception. and I would like to make it more readable in the new commit https://github.com/BoltApp/bolt-magento1/pull/113/commits/d20d831e4561142c5aef605e58f992772f9b47c1

Reply:
To my understanding, do you  want to replace
> 'city'       => @$shippingAddressOriginal['city'],

with 

> 'city'       => isset($shippingAddressOriginal['city']) ? $shippingAddressOriginal['city'] : null,

for better readability? 

I think you should assume whoever works on this repo has decent PHP knowledge, and with that, IMO `@$shippingAddressOriginal['city']` is more readable.